jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1121,6 +1133,22 @@ typedef struct
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+
+/**
+ * Internal properties of TypedArray Object
+ */
+typedef struct
+{
+  /** The arraybuffer object behind the typedarray */
+  ecma_object_t *viewed_arraybuffer_p;
+  /** the byteoffset of the above arraybuffer */
+  ecma_length_t byte_offset;
+  /** the size of each element in the typedarray */
+  ecma_length_t bytes_per_element;
```

Comment:
I suspect this is a read-only property, so at construction time we can decide whether the offset is >= 255, and allocate extra space for it. But I think 0 is the most common value so we don''t need to allocate extra space in most cases.